src/enemies/test_human/TestHuman.gd

```diff
@@ -0,0 +1,107 @@
+extends CharacterBody2D
+## Test class for a human-like enemy.
+
+## Value that dictates how fast this unit can move.
+@export_range(0.0, 1000.0, 0.5, "or_greater", "suffix:px/s") var movement_speed: float = 200.0
+## Value that dictates how much damage an attack from this unit does.
+@export_range(0.0, 100.0, 0.5, "or_greater", "suffix:dmg") var attack_damage: float = 5.0
+## Value that dictates the range at which this unit performs melee attacks. [br]
+## This value ought to be smaller than the [NavigationAgent2D]'s [member NavigationAgent2D.path_desired_distance] value,
+## so that the unit actually moves in close enough to attack.
+@export_range(0.0, 1000.0, 0.5, "or_greater", "suffix:px") var attack_range: float = 150.0
+## Value that dictates how much health this unit spawns in with.
+@export_range(0.0, 200.0, 0.5, "or_greater", "suffix:hp") var max_health: float = 100.0
+## Current health.
+var health: float = max_health
+
+## Value the dictates how long it takes for this unit to attack again after it has already done so.
+@export_range(0.0, 10.0, 0.005, "or_greater", "suffix:s") var attack_delay: float = 2.0
+## Current attack delay, this ticks down during physics process.
+var current_attack_delay: float = 0.0
+
+## This unit's [NavigationAgent2D]. It can't move without one.
+@onready var navigation_agent: NavigationAgent2D = $NavigationAgent2D
+## This unit's [Area2D]. Used for seeing enemies.
+@onready var aggro_area: Area2D = $AggroArea
+
+## This unit's current [CharacterBody2D] target. If it doesn't have one, it will spin until it spots one.
+var target: CharacterBody2D = null
+
+## Value dictating how aware this unit is on the existance of the current target.
+## The value quickly rizes when it spots a target, but it goes slowly down when the target has managed to run away or hide.
+var alert_level: float = 0.0: set = _set_alert_level
+
+func _set_alert_level(new_alert_level: float) -> void:
+	alert_level = clamp(new_alert_level, 0.0, 1.0)
+
+func _process(delta: float) -> void:
+	if target != null:
+		if alert_level > 0.0:
+			if aggro_area.get_overlapping_bodies().has(target):
+				alert_level += delta
```

Comment:
get_overlapping bodies will only update during the physics frame, so this should probably be in a physics process instead of normal process

Reply:
and yeah, i know you're doing animation stuff, but it physically will not change until then